org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeGenerationUtils.java

```diff
@@ -51,10 +52,30 @@ public static IJavaElement findInsertElement(IType type, int currentOffset) {
 		} else if (Objects.equals(insertionLocation, INSERT_BEFORE_CURSOR)) {
 			return findElementAtPosition(type, currentOffset);
 		}
-		
+
 		return findElementAfterPosition(type, currentOffset);
 	}
 
+	public static IJavaElement findInsertElementAfterLastField(IType type) {
+		int lastOffset = 0;
+		try {
+			IJavaElement[] members = type.getChildren();
```

Comment:
Since the results of `IParent.getChildren()` are in no particular order, we have to call `findElementAfterPosition` after we find the last field.